### ISSUE TITLE
Improve Type Hints

### ISSUE THREAD

**Issue by Gonmeso** [gizatechxyz/actions-sdk#36]
Type hinting is inconsistent throughout the code.

We should aim to improve this and make it more consistent and thus, useful in development.

To check which functions have missing hints or hints are not appropriate we can rely on the usage of `mypy`.  We can run the following to have an overview:

```bash
mypy giza_actions --disallow-untyped-defs
```

This will show multiple hinting errors that we should aim to solve. 

We can omit the errors coming from external packages that do not have type hints, for example:

```
giza_actions/utils.py:5: error: Skipping analyzing "giza": module is installed, but missing library stubs or py.typed marker  [import-untyped]```

**Comment by Gerson2102** [comment]
Hello, can I work on this one? Is there any group to ask questions about this issue?

**Comment by raphaelDkhn** [comment]
Hey! Yes here is the Telegram [link](https://t.me/+RLNr83XqyaQ3ZjBk)

**Comment by Gonmeso** [comment]
Fixed in #44
